tests/generation/FileGlobs.py

```diff
@@ -71,14 +74,39 @@ def test_get_project_files_gitignore(self):
             "globexp.py",
             "upload.c",
             os.path.join("src", "main.pyc"),
+            os.path.join("another_folder", "script.js"),
+            os.path.join("data", "sample", "index.html"),
             "run.pyc"]
 
         for file in files + ignored_files:
             os.makedirs(os.path.dirname(os.path.abspath(file)), exist_ok=True)
             open(file, "w").close()
         files += [".gitignore"]
+        os.chdir("another_folder")
+        with open('.gitignore', 'w') as f:
```

Comment:
It would be slightly more intuitive to read up the code if we write these two new  `.gitignore` files just after the first `.gitignore` because you're adding `scripts.js` and `index.html` to ignored files before defining the relevant gitignores.